File: samples/php/smartid_sample.php
<?php
  require(realpath(dirname(__FILE__)).DIRECTORY_SEPARATOR."../../bindings/php/phpSmartIdEngine.php");

function OutputRecognitionResult($recog_result) {
  printf("Document type: %s\n", $recog_result->GetDocumentType());
  printf("Match results:\n");
  $match_results = $recog_result->GetMatchResults();
  for ($i = 0; $i < $match_results->size(); $i++) {
    $match_result_quad = $match_results->get($i)->GetQuadrangle();
    printf("    Template Type = %s\n", $match_results->get($i)->GetTemplateType());
    printf("    Zone = { (%f, %f), (%f, %f), (%f, %f), (%f, %f) }\n", 
        $match_result_quad->GetPoint(0)->x, $match_result_quad->GetPoint(0)->y,
        $match_result_quad->GetPoint(1)->x, $match_result_quad->GetPoint(1)->y,
        $match_result_quad->GetPoint(2)->x, $match_result_quad->GetPoint(2)->y,
        $match_result_quad->GetPoint(3)->x, $match_result_quad->GetPoint(3)->y);
  }

  printf("Segmentation results:\n");
  $segmentation_results = $recog_result->GetSegmentationResults();
  for ($i = 0; $i < $segmentation_results->size(); $i++) {
    $raw_fields_names = $segmentation_results->get($i)->GetRawFieldsNames();
    for ($j = 0; $j < $raw_fields_names->size(); $j++) {
      printf("    Raw field %s\n", $raw_fields_names->get($j));
    } 
  }

  printf("String fields:\n");
  $string_field_names = $recog_result->GetStringFieldNames();
  for ($i = 0; $i < $string_field_names->size(); $i++) {
    $field = $recog_result->GetStringField($string_field_names->get($i));
    $is_accepted = $field->IsAccepted() ? " [+] " : " [-] ";
    printf("    %s\t%s\t%s\n", 
        $field->GetName(), $is_accepted, $field->GetUtf8Value());
  }

  printf("Image fields:\n");
  $image_field_names = $recog_result->GetImageFieldNames();
  for ($i = 0; $i < $image_field_names->size(); $i++) {
    $field = $recog_result->GetImageField($image_field_names->get($i));
    $is_accepted = $field->IsAccepted() ? " [+] " : " [-] ";
    printf("    %s\t%s W: %d H: %d\n", 
        $field->GetName(), $is_accepted,
        $field->GetValue()->getWidth(), $field->GetValue()->getHeight());
  }

  printf("Result terminal: %s\n", $recog_result->IsTerminal() ? " [+] " : " [-] ");
}

function main($argc, $argv) {
  if ($argc != 4) {
    printf("Usage: %s %s <path-to-image-file> <path-to-config-bundle> <document-type>\n", PHP_BINARY, $argv[0]);
    printf("Example: %s %s ../testdata/image.png ../data-zip/bundle_mrz.zip mrz.*\n", PHP_BINARY, $argv[0]);
    exit(-1);
  }

  $image_path = $argv[1];
  $config_path = $argv[2];
  $document_types = $argv[3];

  printf("SmartIDReader version: %s\n", RecognitionEngine::GetVersion());
  printf("image_path = %s\n", $image_path);
  printf("config_path = %s\n", $config_path);
  printf("document_types = %s\n", $document_types);

  try {
    $engine = new RecognitionEngine($config_path);

    $session_settings = $engine->CreateSessionSettings();
    // specify a concrete document type or wildcard mask
    $session_settings->AddEnabledDocumentTypes($document_types);

    //  load image

    $image = new Image();

      //  This works only in trial and full versions

        // $image = new Image($image_path);

    # get file as binary buffer

    $handle = fopen($image_path, "rb");
    $size = filesize($image_path);
    $contents = fread($handle, $size);
    fclose($handle);

    # get file as Base64 string
    
    $handleForBase64 = fopen(realpath(dirname(__FILE__)).DIRECTORY_SEPARATOR."base64.txt", "r");
    $sizeForBase64 = filesize($image_path);
    $base64string = fgets($handleForBase64);
    fclose($handleForBase64);

    # You can take base64 buffer directly from Image object

    // $base64stringInternal = $image->GetBase64String();

    $session = $engine->SpawnSession($session_settings, $optional_reporter); 

    // Uses engine's internal image loading, supports format: png, jpg, jpeg, tif
    $resultImage = $session->ProcessImage($image);
    $session->Reset();

    $resultFile = $session->ProcessImageFile($image_path);
    $session->Reset();
    
    $resultData = $session->ProcessImageData($contents, $size);
    $session->Reset();
    
    $resultBase64 = $session->ProcessImageDataBase64($base64string);
    $session->Reset();
    
    // $resultBase64Internal = $session->ProcessImageDataBase64($base64stringInternal);

    OutputRecognitionResult($resultImage);
    OutputRecognitionResult($resultFile);
    OutputRecognitionResult($resultData);
    OutputRecognitionResult($resultBase64);
    // OutputRecognitionResult($resultBase64Internal);

  } catch (Exception $e) {
    printf("Exception caught: %s\n", $e->getMessage());
    exit(-2);
  }
}

main($argc, $argv);

?>

File: samples/python/smartid_sample.py
import sys
import os

sys.path.append(os.path.join(sys.path[0], '../../bindings/python/'))
sys.path.append(os.path.join(sys.path[0], '../../bin/'))

import pySmartIdEngine


def output_recognition_result(result):
    print('Document type: %s' % result.GetDocumentType())
    print('Match results:')
    match_results = result.GetMatchResults()
    for i in range(len(match_results)):
        match_result_quad = match_results[i].GetQuadrangle()
        print('    Template Type = %s' % match_results[i].GetTemplateType())
        print('    Zone = { (%f, %f), (%f, %f), (%f, %f), (%f, %f) }' % (
            match_result_quad.GetPoint(0).x, match_result_quad.GetPoint(0).y,
            match_result_quad.GetPoint(1).x, match_result_quad.GetPoint(1).y,
            match_result_quad.GetPoint(2).x, match_result_quad.GetPoint(2).y,
            match_result_quad.GetPoint(3).x, match_result_quad.GetPoint(3).y))

    print('Segmentation results:')
    segmentation_results = result.GetSegmentationResults()
    for i in range(len(segmentation_results)):
        raw_fields_names = segmentation_results[i].GetRawFieldsNames()
        for j in range(len(raw_fields_names)):
            print('    Raw field %s' % raw_fields_names[j])

    print('String fields:')
    string_field_names = result.GetStringFieldNames()
    for i in range(len(string_field_names)):
        field = result.GetStringField(string_field_names[i])
        is_accepted = ' [+] ' if field.IsAccepted() else ' [-] '
        print('    %-20s %s %s' % (field.GetName(), is_accepted, field.GetUtf8Value()))

    print('Image fields:')
    image_field_names = result.GetImageFieldNames()
    for i in range(len(image_field_names)):
        field = result.GetImageField(image_field_names[i])
        is_accepted = ' [+] ' if field.IsAccepted() else ' [-] '
        print('    %-20s %s W: %d H: %d' % (field.GetName(), is_accepted,
                                            field.GetValue().GetWidth(), field.GetValue().GetHeight()))

    print('Result terminal:         %s' % (' [+] ' if result.IsTerminal() else ' [-] '))


def main():
    if len(sys.argv) != 4:
        print('Usage: %s <path-to-image-file> <path-to-config-bundle> <document-type>' % sys.argv[0])
        print('Example: %s ../testdata/image.png ../data-zip/bundle_mrz.zip mrz.*' % sys.argv[0])
        sys.exit(-1)

    image_path, config_path, document_types = sys.argv[1:]

    print('RecognitionEngine Version: %s' % pySmartIdEngine.RecognitionEngine.GetVersion())
    print('Creating RecognitionEngine from %s' % config_path)

    engine = pySmartIdEngine.RecognitionEngine(config_path)
    
    session_settings = engine.CreateSessionSettings()
    set_enabled_document_types = pySmartIdEngine.StringVector()
    set_enabled_document_types.append(document_types)
    session_settings.SetEnabledDocumentTypes(set_enabled_document_types)

# load image 

    testImage = pySmartIdEngine.Image()

  # This works only in trial and full versions

    # testImage = pySmartIdEngine.Image(image_path) 

# get file as binary buffer 

    binaryRGBfile = open(image_path, "rb")
    size = os.path.getsize(image_path)
    binaryRGBfileBuffer = bytearray(size)
    binaryRGBfile.readinto(binaryRGBfileBuffer)

# get raw RGB data from file 

  # You can take image buffer directly from Image object, works in trial and full versions

    # bufferSizeForRGB = testImage.GetHeight()*testImage.GetStride()*testImage.GetChannels()
    # bufferForRGB = bytearray(bufferSizeForRGB)
    # testImage.CopyToBuffer(bufferForRGB, bufferSizeForRGB)

# get file as Base64 string

    f = open(os.path.join(sys.path[0], "base64.txt"), 'r')
    base64BufferString = f.readline()
    f.close()

  # You can take base64 buffer directly from Image object

    # base64BufferSize = testImage.GetRequiredBase64BufferLength()  
    # base64Buffer = bytearray(base64BufferSize)
    # testImage.CopyBase64ToBuffer(base64Buffer, base64BufferSize)
    # base64BufferString = base64Buffer.decode()
    # base64BufferString = str(base64BufferString) #explicit conversion need for Python 2

  # or you can take base64 string directly by
    
    # base64BufferString = testImage.GetBase64String()

# start recognition session

    session = engine.SpawnSession(session_settings)

    resultFromImage = session.ProcessImage(testImage)
    session.Reset()

    resultFromImageFile = session.ProcessImageFile(image_path)
    session.Reset()

    resultFromImageData = session.ProcessImageData(binaryRGBfileBuffer, size)
    session.Reset()

    resultFromBase64 = session.ProcessImageDataBase64(base64BufferString)
    session.Reset()

    # resultFromRawBuffer = session.ProcessSnapshot(bufferForRGB, bufferSizeForRGB, testImage.GetWidth(), testImage.GetHeight(), testImage.GetStride(), testImage.GetChannels()) 

# print result

    output_recognition_result(resultFromImage)
    output_recognition_result(resultFromImageFile)
    output_recognition_result(resultFromImageData)
    output_recognition_result(resultFromBase64)
    # output_recognition_result(resultFromRawBuffer)



if __name__ == '__main__':
    main()


